### PR TITLE
Somnokid

### PR DESCRIPTION
**Pull request by slobters**
# New Port for Somnokid

## Game Information
- **Title**: Somnokid
- **URL**: https://team-bugulon.itch.io/somnokid

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [ ] 480x320
- [X] 640x480
- [ ] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/somnokid/Somnokid.sh
Comment (slobters) [inline ports/somnokid/Somnokid.sh:1]: Removed "TOOLDIR="$GAMEDIR/tools" " and line at the end

File: ports/somnokid/Somnokid.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+$ESUDO chmod 777 "$GAMEDIR/gmloadernext"
+
+# Splash
+[ "$CFW_NAME" == "muOS" ] && splash "splash.png" 1 # workaround for muOS
+$ESUDO ./tools/splash "splash.png" 10000 &
+
+$GPTOKEYB "gmloadernext" -c "controls.gptk" &
+pm_platform_helper $GAMEDIR/gmloadernext
```
Comment (cdeletre) [inline ports/somnokid/Somnokid.sh:41]:
```suggestion
pm_platform_helper "$GAMEDIR/gmloadernext"
```

File: ports/somnokid/Somnokid.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+cd $GAMEDIR
+
+# We log the execution of the script into log.txt
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+$ESUDO chmod 777 "$GAMEDIR/gmloadernext"
+
+# Splash
+[ "$CFW_NAME" == "muOS" ] && splash "splash.png" 1 # workaround for muOS
+$ESUDO ./tools/splash "splash.png" 10000 &
```
Comment (cdeletre) [inline ports/somnokid/Somnokid.sh:38]: `splash` is in the `PATH` :)
```suggestion
$ESUDO splash "splash.png" 10000 &
```

File: ports/somnokid/Somnokid.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+# Variables
+GAMEDIR="/$directory/ports/somnokid"
+TOOLDIR="$GAMEDIR/tools"
+
+# Exports
+export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH"
```
Comment (cdeletre) [inline ports/somnokid/Somnokid.sh:25]: As discussed with Cebion in [PR 818](https://github.com/PortsMaster/PortMaster-New/pull/818#discussion_r1808166816) it's better and easier in the long run.

Don't forget to also rename the `libs` folder to `libs.aarch64`

```suggestion
export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/libs.${DEVICE_ARCH}:$LD_LIBRARY_PATH"
```

**Review by cdeletre — CHANGES_REQUESTED** [review]
You can also add these licenses:
* [LICENSE.libcrypto.txt](https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libcrypto.txt)
* [LICENSE.libopenal.txt](https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libopenal.txt)
* [LICENSE.libzip.txt](https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libzip.txt)

**Review by Cebion — APPROVED** [review]

**Review by cdeletre — APPROVED** [review]

**Pull request merged**